docs/source/array-creation.rst

```diff
@@ -201,6 +201,42 @@ on your Dask array to turn in to a normal NumPy array:
    array([0, 1, 4, 9, 16, 25])
 
 
+Intermediate storage
+--------------------
```

Comment:
I think that this section should come after sections like numpy-style-slicing, hdf5, zarr, etc..  I think that those are relatively more common.

Reply:
Not sure. If we are going to add a section on `persist` and then on `compute(..., return_stored=True)` (as suggested below), then it probably makes sense where it is as we are enumerating different ways of generally computing data. It would be a little strange to list some forms of computation, some storage formats, and then some other forms of computation. That is just my opinion though.

Reply:
I think that we should probably add a small section to the In Memory section just above that includes `persist` (my guess is that we wrote those docs before persist).  I think that we should expand the title of the Intermediate Storage section to say "Intermediate Storage on Disk" or something similar.

Similarly this sentence

> In some cases, one may wish to store an intermediate value that will be used in later computations

Might be expanded to include the following

```diff
-In some cases, one may wish to store an intermediate value that will be used in later computations
+In some cases, one may wish to store an intermediate value to disk that will be used in later computations
```

Reply:
Any other thoughts on this?

Reply:
I still think that this section is less relevant than storing to HDF5 or Zarr, which I have found come up much more frequently.  This seems to be more of an advanced technique (though I agree very useful).  I think that the other sections are simpler and more commonly applicable, and so should come first.

Reply:
Thanks @jakirkham !

Reply:
It's been moved.